contrib/drivers/lvm/lvm_metrics.go

```diff
@@ -127,67 +117,107 @@ func (d *MetricDriver) ValidateMetricsSupportList(metricList []string, resourceT
 	for _, resources := range configs.Cfgs {
 		if resources.Resource == resourceType {
 			switch resourceType {
-			//ToDo: Other Cases needs to be added
 			case "volume", "disk":
-				for _, metricName := range metricList {
-					if metricInMetrics(metricName, resources.Metrics) {
-						supportedMetrics = append(supportedMetrics, metricName)
+				for _, m := range resources.Metrics {
+					supportedMetrics = append(supportedMetrics, m)
 
-					} else {
-						log.Infof("metric:%s is not in the supported list", metricName)
-					}
 				}
 			}
 		}
 	}
+
 	return supportedMetrics, nil
 }
 
 //	CollectMetrics: Driver entry point to collect metrics. This will be invoked by the dock
-//	metricsList-> posted metric list
-//	instanceID -> posted instanceID
-//	metricArray	-> the array of metrics to be returned
-func (d *MetricDriver) CollectMetrics(metricsList []string, instanceID string) ([]*model.MetricSpec, error) {
+//	[]*model.MetricSpec	-> the array of metrics to be returned
+func (d *MetricDriver) CollectMetrics() ([]*model.MetricSpec, error) {
 
 	// get Metrics to unit map
 	metricToUnitMap := getMetricToUnitMap()
 	//validate metric support list
-	supportedMetrics, err := d.ValidateMetricsSupportList(metricsList, "volume")
+	supportedMetrics, err := d.GetMetricList("volume")
 	if supportedMetrics == nil {
 		log.Infof("no metrics found in the  supported metric list")
 	}
-	metricMap, labelMap, err := d.cli.CollectMetrics(supportedMetrics, instanceID)
-
+	// discover lvm volumes
+	volumeList, err := d.cli.DiscoverVolumes()
+	// discover lvm physical volumes
+	DiskList, err := d.cli.DiscoverDisks()
+	metricMap, labelMap, err := d.cli.CollectMetrics(supportedMetrics)
```

Comment:
This err will override the last two err variables before, if you just want to validate this error, please use `_` instead for last two.

Reply:
> This err will override the last two err variables before, if you just want to validate this error, please use `_` instead for last two.

Thanks for pinpoint, I have handled this